esmvalcore/preprocessor/_volume.py

```diff
@@ -182,6 +184,8 @@ def calculate_volume(cube: Cube) -> da.core.Array:
         thickness, cube.shape, z_dim, chunks=chunks
     )
     grid_volume = area_arr * thickness_arr
+    if cube.has_lazy_data():
+        grid_volume = grid_volume.rechunk(chunks)
```

Comment:
Does this do anything? I would expect chunks to be preserved on multiplication.

Reply:
I would expect so too, but this probably can't hurt? It looks like `rechunk` is a no-op if the chunks already match:

```py
import dask.array as da
x = da.ones((12, 24, 48), chunks=(6, 12, 24))
y = da.ones((12, 24, 48), chunks=(6, 12, 24)).rechunk((6, 12, 24))
print(y.dask == x.dask)  # True
```

Reply:
I thought this would always be the case since we're broadcasting with the chunks specified, but as you pointed out, this will only rechunk dimensions that are being broadcast, so you're right and it does make sense to have this here.

Reply:
Technically, the non-broadcasted arrays `thickness` and `area_array` should already have the correct chunks (this is ensured in our code/Iris code earlier), but better safe than sorry.